app/server/methods/activityReports.js

```diff
@@ -120,3 +129,226 @@ export const mergeHomes = (dataRows, homeIds) => {
     ).map((val) => ({ key: val[0], value: val[1] })),
   }));
 };
+
+export const totalResidentsPerHome = (endDate, periodInDays) => {
+  const homes = Homes.find().fetch();
+  const residents = Residencies.find({
+    $and: [
+      {
+        moveIn: { $lte: new Date(endDate) },
+      },
+      {
+        $or: [
+          { moveOut: { $exists: false } },
+          { moveOut: { $gte: new Date(endDate) } },
+        ],
+      },
+    ],
+  }).fetch();
+
+  return homes.reduce((acc, current) => {
+    return {
+      ...acc,
+      [current._id]: {
+        name: current.name,
+        residents: getResidentsForEachDay(
+          residents,
+          endDate,
+          periodInDays
+        ),
+      },
+    };
+  }, {});
+};
+
+export const getResidentsForEachDay = (
+  residents,
+  endDate,
+  periodInDays
+) => {
+  const startDate = moment(endDate)
+    .clone()
+    .subtract('days', periodInDays);
+  const dates = enumerateDaysBetweenDates(startDate, endDate);
+  return dates.map((date) => ({
+    date,
+    residents: residents.filter(
+      (resident) =>
+        new Date(resident.moveIn).getTime() <=
+          new Date(date).getTime() &&
+        (!resident.moveOut
+          ? true
+          : new Date(resident.moveOut).getTime() >
+            new Date(date).getTime())
+    ),
+  }));
+};
+
+export const enumerateDaysBetweenDates = (startDate, endDate) => {
+  const dates = [];
+
+  const currDate = moment(startDate).startOf('day');
+  const lastDate = moment(endDate).startOf('day');
+
+  while (currDate.add(1, 'days').diff(lastDate) < 0) {
+    dates.push(currDate.clone().toDate());
+  }
+
+  return dates;
+};
+
+export const totalActiveResidentsPerHome = (
+  endDate,
+  periodInDays
+) => {
+  const startDate = moment(endDate)
+    .clone()
+    .subtract('days', periodInDays);
+  const activities = getActivityWithHomes(startDate, endDate);
+  const homeActivityMap = activities.reduce((homeMap, current) => {
+    if (!homeMap[current.homeId]) {
+      homeMap[current.homeId] = {
+        residentActivities: [],
+      };
+    }
+
+    homeMap[current.homeId].residentActivities.push(current);
+    return homeMap;
+  }, {});
+
+  Object.keys(homeActivityMap).forEach((home) => {
+    const dates = enumerateDaysBetweenDates(startDate, endDate);
+
+    homeActivityMap[home].residents = dates.map((date) => ({
+      date,
+      residents: homeActivityMap[home].residentActivities.filter(
+        (resident) => {
+          return (
+            moment(resident.activityDate).format('YYYY-MM-DD') ===
+            moment(date).format('YYYY-MM-DD')
+          );
+        }
+      ),
+    }));
+  });
+  return homeActivityMap;
+};
+
+/**
```

Comment:
Add a general description to this docstring along with descriptions of input paramaters.